src/adagrams.js

```diff
@@ -1,15 +1,125 @@
+const letterPool = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  let letterBank = [];
+  let letterHand = [];
+  {
+    for (const [letter, count] of Object.entries(letterPool)) {
+      for (let i = 0; i < count; i++) {
+        letterBank.push(letter);
+      }
+    }
+    for (let i = 0; i < 10; i++) {
+      let randomIndex = Math.floor(Math.random() * letterBank.length);
+      let randLetter = letterBank[randomIndex];
+      letterHand.push(randLetter);
+      letterBank.splice(randomIndex, 1);
+    }
+    return letterHand;
+  }
 };
+console.log(drawLetters());
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let inputUpper = input.toUpperCase();
+  let letterInHandCopy = [...lettersInHand];
+  for (let letter of inputUpper) {
+    if (letterInHandCopy.includes(letter)) {
+      let indexHand = letterInHandCopy.indexOf(letter);
+      letterInHandCopy.splice(indexHand, 1);
+    } else {
+      return false;
+    }
+  }
+  return true;
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  if (!word ){
+    return 0
+  }
+  let letterValsObj = {
```

Comment:
This value isn't reassigned so it should be `const` instead of `let`. I'd also recommend moving this outside the function into the main module scope so it isn't recreated every time `scoreWord` is called.